### PR TITLE
PMM-9312 Implemented collstats & indexstats limits

### PR DESCRIPTION
**Pull request by percona-csalguero**
[PMM-XXXX](https://jira.percona.com/browse/PMM-XXXX)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/0

- [ ] Links to other linked pull requests (optional).
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: services/agents/mongodb_test.go [hunk 1]
```diff
@@ -49,10 +49,8 @@ func TestMongodbExporterConfig225(t *testing.T) {
 		TemplateLeftDelim:  "{{",
 		TemplateRightDelim: "}}",
 		Args: []string{
-			// temporarily disabled by PMM-9304 until we have more information about the resources
-			// consumption for these 2 collectors.
-			// "--collector.dbstats",
-			// "--collector.topmetrics",
+			"--collector.diagnosticdata",
+			"--collector.replicasetstatus",
```
Comment (BupycHuk) [inline services/agents/mongodb_test.go:53]: if I got it right these two flags were added recently, so we should handle version.
different flags should be used for 2.25 and 2.26

File: services/agents/mongodb_test.go [hunk 2]
```diff
@@ -89,6 +85,55 @@ func TestMongodbExporterConfig225(t *testing.T) {
...
+			"--discovering-mode",
+			"--mongodb.collstats-colls=col1,col2,col3",
+			"--mongodb.global-conn-pool",
+			"--web.listen-address=:{{ .listen_port }}",
+		}
+		actual := mongodbExporterConfig(mongodb, exporter, exposeSecrets, pmmAgentVersion)
+		require.Equal(t, expected.Args, actual.Args)
+	})
+
+	t.Run("Enabling all collectors but collstats-limit=0", func(t *testing.T) {
```
Comment (BupycHuk) [inline services/agents/mongodb_test.go:114]: please add test for pmm client 2.25

**Review by JiriCtvrtka — APPROVED** [review]
Please add version check

**Review by BupycHuk — CHANGES_REQUESTED** [review]
both of my comments weren't addressed, until we fix issue with different flags for different versions I can't approve this PR.

File: services/agents/mongodb.go [hunk 3]
```diff
@@ -160,6 +211,39 @@ func v225Args(exporter *models.Agent, disabledCollectors []string, tdp *models.D
...
+			enabled:     true,
+			enableParam: "--collector.diagnosticdata",
+		},
+		"replicasetstatus": {
+			enabled:     true,
+			enableParam: "--collector.replicasetstatus",
+		},
+		// disabled until we have better information on the resources usage impact
+		"collstats": {
+			enabled:     false || (collectAll && collstatsLimit != 0),
```
Comment (BupycHuk) [inline services/agents/mongodb.go:226]: false || is uneccesary here and in cases below.

Comment (percona-csalguero) [reply services/agents/mongodb.go]: Fixed. Could you re -review?

File: services/agents/mongodb.go [hunk 4]
```diff
@@ -99,7 +107,50 @@ func mongodbExporterConfig(service *models.Service, exporter *models.Agent, reda
...
+
+	collstatsLimit := int32(200)
+	if exporter.MongoDBOptions != nil && exporter.MongoDBOptions.CollectionsLimit != -1 {
+		collstatsLimit = exporter.MongoDBOptions.CollectionsLimit
+	}
+
+	collectors := defaultCollectors(collectAll, collstatsLimit)
+
+	for _, collector := range exporter.DisabledCollectors {
+		col := collectors[strings.ToLower(collector)]
```
Comment (BupycHuk) [inline services/agents/mongodb.go:124]: make sure that collector is exist in the map.

**Review by BupycHuk — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**